src/test/java/org/motechproject/ebodac/osgi/EbodacEnrollmentServiceIT.java

```diff
@@ -0,0 +1,817 @@
+import org.motechproject.testing.osgi.container.MotechNativeTestContainerFactory;
+import org.ops4j.pax.exam.ExamFactory;
+import org.ops4j.pax.exam.junit.PaxExam;
+import org.ops4j.pax.exam.spi.reactors.ExamReactorStrategy;
+import org.ops4j.pax.exam.spi.reactors.PerSuite;
+import org.osgi.framework.BundleContext;
+import org.quartz.JobDetail;
+import org.quartz.JobKey;
+import org.quartz.Scheduler;
+import org.quartz.SchedulerException;
+import org.quartz.Trigger;
+import org.quartz.TriggerKey;
+import org.quartz.impl.matchers.GroupMatcher;
+import org.quartz.impl.matchers.NameMatcher;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+import static junit.framework.Assert.assertEquals;
+import static org.motechproject.commons.date.util.DateUtil.newDateTime;
+import static org.motechproject.testing.utils.TimeFaker.fakeNow;
+import static org.motechproject.testing.utils.TimeFaker.stopFakingTime;
+
+@RunWith(PaxExam.class)
+@ExamReactorStrategy(PerSuite.class)
+@ExamFactory(MotechNativeTestContainerFactory.class)
+public class EbodacEnrollmentServiceIT extends BasePaxIT {
+
+    @Inject
+    private BundleContext bundleContext;
+
+    @Inject
+    EbodacEnrollmentService ebodacEnrollmentService;
+
+    @Inject
+    SubjectService subjectService;
+
+    @Inject
+    VisitService visitService;
+
+    @Inject
+    MessageCampaignService messageCampaignService;
+
+    @Inject
+    EnrollmentDataService enrollmentDataService;
+
+    @Inject
+    SubjectEnrollmentsDataService subjectEnrollmentsDataService;
+
+    @Inject
+    RaveImportService raveImportService;
+
+    Scheduler scheduler;
+
+    private final static String campaignCompletedString = "org.motechproject.messagecampaign.campaign-completed-EndOfCampaignJob";
+    private final static String firedCampaignString = "org.motechproject.messagecampaign.fired-campaign-message-MessageJob";
+    private final static String runonce = "-runonce";
+
+    @Before
+    public void cleanBefore() throws SchedulerException {
+        scheduler = (Scheduler) getQuartzScheduler(bundleContext);
+        clearJobs();
+        subjectEnrollmentsDataService.deleteAll();
+        enrollmentDataService.deleteAll();
+        subjectService.deleteAll();
+    }
+
+    @After
+    public void cleanAfter() throws SchedulerException {
+        clearJobs();
+        subjectEnrollmentsDataService.deleteAll();
+        enrollmentDataService.deleteAll();
+        subjectService.deleteAll();
+    }
+
+    @Test
+    public void shouldCreateEnrollments() throws IOException {
+        try {
+            fakeNow(newDateTime(2015, 8, 1, 0, 0, 0));
+
+            Subject subject = createSubjectWithRequireData("1");
+            subjectService.createOrUpdateForZetes(subject);
```

Comment:
Is any reason why you using createOrUpdateForZetes not just create? If no you can add this to createSubjectWithRequireData method